### ISSUE TITLE
meta: Cargo.lock is missing?

### ISSUE THREAD

**Issue by amadalusia** [doukutsu-rs/doukutsu-rs#273]
I was checking about the repository because I was going to make a package for it in Nix, although something peculiar I noted was the lack of a `Cargo.lock` file. While you are aware that the `Cargo.lock` was removed, and on purpose it seems, I am not really certain if you know why it is supposed to be included. It is never seen for a Cargo-based project to not include a `Cargo.lock` file.

Let's introduce 2 characters here: Xavier and Isabela.

Isabela has an issue with the game because of a dependency that for whatever reason, has had a change that broke doukutsu-rs.
Xavier will only know what she means if he either:

- Xavier just so happened to install doukutsu-rs at just the right time where the dependencies are the exact same for one another
- or Xavier gets a copy of Isabela's `Cargo.lock` file so they both get the exact same dependency hashes.

This also means that developers may not be running the same dependency version as another developer, whereas one may have a method, structure, whatever that has been deprecated and another may have a new killer feature. Since both would have to create a lock file.

In conclusion: the `Cargo.lock` file should not be ignored because it ensures that all developers and end-users have the exact dependencies that the game can work on.

Otherwise, your efforts in maintaining this is well appreciated. Cheers!

**Comment by alula** [comment]
hmm... I just noticed this in our .gitignore and this seems like a bit of bullshit, was carried over from GitHub's Rust .gitignore template:

```
# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
Cargo.lock
```

**Comment by amadalusia** [comment]
That's actually quite funny that GitHub automatically ignores this! :rofl:

Sorry for making assumptions!

**Comment by alula** [comment]
This apparently used to be an "official recommendation for binaries" and was added around 9 years ago...

https://github.com/github/gitignore/pull/1752

**Comment by amadalusia** [comment]
lol

think its worth locking at this point, my issue is now gone (though i do have to hold up a fork for the time being)

**Comment by amadalusia** [comment]
also just noticed that this also affects every rust repository made by the doukutsu-rs account :sweat_smile: 
ill be glad to make the prs

**Comment by alula** [comment]
I think that adding Cargo.lock to every git pulled dependency is unnecessary.

Everything including Git dependencies is locked in doukutsu-rs Cargo.lock: https://github.com/doukutsu-rs/doukutsu-rs/blob/master/Cargo.lock

The separate git repositories are not intended to be built standalone, and seems like the consensus about Cargo.lock is "commit it only for applications".

I think having Cargo.lock just in doukutsu-rs is enough unless proven otherwise.
